emcc.py

```diff
@@ -186,12 +188,13 @@ def run():
     debug_env = os.environ.copy()
     debug_env['EMCC_DEBUG'] = '1'
     args = filter(lambda x: x != '--cflags', sys.argv)
-    out, err = subprocess.Popen([shared.PYTHON] + args + [shared.path_from_root('tests', 'hello_world.c'), '-c'], stderr=subprocess.PIPE, env=debug_env).communicate()
-    lines = filter(lambda x: 'running:' in x and 'clang' in x, err.split(os.linesep))
-    assert len(lines) == 1
-    parts = lines[0].split(' ')[2:]
-    parts = filter(lambda x: x != '-c' and x != '-o' and 'hello_world' not in x and '-emit-llvm' not in x, parts)
-    print ' '.join(parts)
+    with misc_temp_files.get_file(suffix='.o') as temp_target:
+      out, err = subprocess.Popen([shared.PYTHON] + args + [shared.path_from_root('tests', 'hello_world.c'), '-c', '-o', temp_target], stderr=subprocess.PIPE, env=debug_env).communicate()
+      lines = filter(lambda x: 'running:' in x and 'clang' in x, err.split(os.linesep))
```

Comment:
It would probably be more reliable to search for the file name (hello_world.c or whatever, along with 'clang' and maybe '-c' or '-o') rather than depending on the particular format of the debug output 'running:'.